main.cpp

```diff
@@ -42,16 +42,61 @@ struct Point
     }
 private:
     float x{0}, y{0};
-};
+};
+
+template<typename Type>
+struct Wrapper
+{
+    Wrapper(Type&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    Type val;
+
+    void print()
+    {
+        std::cout << val << std::endl;
+    }
+};
+
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    Point val;
+
+    void print()
+    {
+        std::cout << val.toString() << std::endl;
+    }
+};
```

Comment:
Project 9.1 Challenge 1:
There is a lot of repetition between the template class `Wrapper`, and the template specialization for `Wrapper<Point>`
Can you figure out how to eliminate the duplicate constructor?
hint: you only have to _explicitly specialize_ that which is unique.
You can specialize member functions without specializing the entire class :wink: